File: Busqueda lineal 01-09-25/busqueda_lineal.js
const readline = require("readline");

function imprimir(arre) {
    console.log("El arreglo es:", arre.join(" "));
}

function insertar_elemento(arre, val, indice) {
    for (let i = arre.length - 1; i > indice; i--) {
        arre[i] = arre[i - 1];
    }
    arre[indice] = val;
}

function busqueda_lineal(arre, obj) {
    for (let i = 0; i < arre.length; i++) {
        if (arre[i] === obj) {
            return i;
        }
    }
    return -1;
}

let arre = [1, 2, 3, 4, 5];
imprimir(arre);

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Valor para insertar: ", (val) => {
    val = parseInt(val);
    rl.question(`Indice donde se va a insertar (0..${arre.length - 1}): `, (indice) => {
        indice = parseInt(indice);

        insertar_elemento(arre, val, indice);
        imprimir(arre);

        let obj = val;
        let pos = busqueda_lineal(arre, obj);
        console.log(`Busqueda lineal de ${obj}: indice ${pos}`);

        rl.question("\nPresiona enter para salir", () => rl.close());
    });
});

File: Busqueda lineal 01-09-25/busqueda_lineal.py
def imprimir(arre):

    print("El arreglo inicial es:", " ".join(str(x) for x in arre))


def insertar_elemento(arre, val, indice):

    n = len(arre)
    for i in range(n - 1, indice, -1):
        arre[i] = arre[i - 1]

    arre[indice] = val


def busqueda_lineal(arre, obj):
 
    for i in range(len(arre)):
        if arre[i] == obj:
            return i
    return -1


arre = [1, 2, 3, 4, 5]
imprimir(arre)

val = int(input("Valor para insertar: "))
indice = int(input("Indice donde se va a insertar (0..{}): ".format(len(arre)-1)))

insertar_elemento( arre, val, indice)
imprimir(arre)

obj = val
pos = busqueda_lineal(arre, obj)
print(f"Búsqueda lineal de {obj}: indice {pos}")
input("\nPresiona enter para salir")
